src/averageAges.js

```diff
@@ -55,7 +75,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrens = onlyWithSon
+    ? people
+      .filter((person) => person.sex === 'm'
+        && people.some((mom) => mom.name === person.mother))
+    : people
+      .filter((person) => people.some((mom) => mom.name === person.mother));
+
+  const mothers = people.filter((mother) =>
+    childrens.find((child) => mother.name === child.mother)
+  );
+
+  const childAgesSum = childrens
+    .map((child) => child.born)
+    .reduce((age, currentAge) => age + currentAge, 0);
+
+  const mothersAge = childrens.map((child) =>
+    mothers
+      .filter((mom) => mom.name === child.mother)
+      .map((age) => age.born)
+      .reduce((age, curent) => age + curent)
```

Comment:
Check all methods here, some of them are redundant
Don't create additional iterators
For example, you can calculate needed age value in reduce without map method